File: app/src/main/java/com/example/android/testing/espresso/BasicSample/test2/Test2ListAdapter.java
package com.example.android.testing.espresso.BasicSample.test2;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.android.testing.espresso.BasicSample.R;

import java.util.List;

public class Test2ListAdapter extends RecyclerView.Adapter<Test2ListAdapter.ViewHolder> {

    private final List<String> items;

    Test2ListAdapter(List<String> items) {
        this.items = items;
    }

    @Override
    public Test2ListAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View itemLayoutView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_layout, null);

        // create ViewHolder
        return new Test2ListAdapter.ViewHolder(itemLayoutView);
    }

    @Override
    public void onBindViewHolder(Test2ListAdapter.ViewHolder holder, int position) {
        holder.vTitle.setText(items.get(position));
    }

    // inner class to hold a reference to each item of RecyclerView
    public static class ViewHolder extends RecyclerView.ViewHolder {

        public TextView vTitle;

        public ViewHolder(View itemLayoutView) {
            super(itemLayoutView);
            vTitle = (TextView) itemLayoutView.findViewById(R.id.test2_text);
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }
}

File: app/src/main/java/com/example/android/testing/espresso/BasicSample/test2/Test2ListAdapter.kt
package com.example.android.testing.espresso.basicSample.test2

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.example.android.testing.espresso.basicSample.R

class Test2ListAdapter (private val items: List<String>) : RecyclerView.Adapter<Test2ListAdapter.ViewHolder>() {

    override fun getItemCount() = items.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Test2ListAdapter.ViewHolder {
        val itemLayoutView = LayoutInflater.from(parent.context).inflate(R.layout.item_layout, null)

        // create ViewHolder
        return Test2ListAdapter.ViewHolder(itemLayoutView)
    }

    override fun onBindViewHolder(holder: Test2ListAdapter.ViewHolder, position: Int) {
        holder.vTitle.text = items[position]
    }

    // inner class to hold a reference to each item of RecyclerView
    class ViewHolder(itemLayoutView: View) : RecyclerView.ViewHolder(itemLayoutView) {
        var vTitle = itemLayoutView.findViewById<View>(R.id.test2_text) as TextView
    }
}
